Src/MoneyFox.Windows.Tasks/LiveTilesTaskService.cs

```diff
@@ -0,0 +1,1253 @@
+			List<Payment> allpayments = new List<Payment>();
+			List<PaymentEntity> newlst = new List<PaymentEntity>();
+			List<RecurringPaymentEntity> allrecurring = new List<RecurringPaymentEntity>();
+			foreach (Account item in acct)
+			{
+				allpayments.AddRange(await paymentService.GetPaymentsByAccountId(item.Data.Id));
+			}
+			foreach (Payment item in allpayments)
+			{
+				if (item.Data.IsRecurring)
+				{
+					Getrecurancepayments(item, item.Data.RecurringPayment.Recurrence, "last");
+				}
+				else
+				{
+					allpayment.Add(item.Data.Date, item);
+				}
+			}
+			var tet = alldates.OrderBy(x => x.Date).TakeLast(8);
+			List<string> returnlist = new List<string>();
+			foreach (var item in tet)
+			{
+				Payment p = allpayment[item];
+				returnlist.Add(string.Format(GetResourceString("LiveTilePastPaymentText"),p.Data.Amount,p.Data.ChargedAccount.Name,p.Data.Date));
+			}
+			alldates.Clear();
+			allpayment.Clear();
+			return returnlist;
+
+		}
+		private async Task<List<string>> Get9NextpaymentsAsync()
+		{
+			var acct = await accountService.GetAllAccounts();
+			DateTime today = DateTime.Now;
+			List<Payment> allpayments = new List<Payment>();
+			List<PaymentEntity> newlst = new List<PaymentEntity>();
+			List<RecurringPaymentEntity> allrecurring = new List<RecurringPaymentEntity>();
+			foreach (Account item in acct)
+			{
+				allpayments.AddRange(await paymentService.GetPaymentsByAccountId(item.Data.Id));
+			}
+			foreach (Payment item in allpayments)
+			{
+				if (item.Data.IsRecurring)
+				{
+					Getrecurancepayments(item, item.Data.RecurringPayment.Recurrence, "next");
+
+				}
+				else
+				{
+					newlst.Add(item.Data);
+				}
+			}
+			List<Payment> sortedlist = new List<Payment>();
+			var te = newlst.OrderBy(x => DateTime.Compare(today, x.Date) <= 0).Take(9);
+			var tes = allrecurring.OrderBy(x => DateTime.Compare(today, x.StartDate) <= 0 && DateTime.Compare(today, DateTime.Parse(x.EndDate.HasValue ? x.EndDate.ToString() : null)) <= 0).Take(9);
+			List<PaymentEntity> test = (List<PaymentEntity>)te;
+			foreach (var item in tes)
+			{
+				PaymentEntity payments = new PaymentEntity();
+				payments.RecurringPayment = item;
+				payments.Date = DateTime.Parse(DateTime.Now.Month + "/" + item.StartDate.Day + "/" + DateTime.Now.Year);
+				payments.Amount = item.Amount;
+				test.Add(payments);
+			}
+			var tested = test.OrderBy(x => x.Date);
+			var tet = tested.Take(9);
+			List<string> returnlist = new List<string>();
+			foreach (var item in tet)
+			{
+				returnlist.Add(string.Format(GetResourceString("LiveTileFuturePaymentText"),item.Amount,item.ChargedAccount.Name,item.Date));
+			}
+
+			return returnlist;
+
+		}
+		private async Task<double> GetLatestBalanceAsync(int tileid)
+		{
+			var te = await accountService.GetById(tileid);
+			return te.Data.CurrentBalance;
+		}
+		private void Getrecurancepayments(Payment p, Foundation.PaymentRecurrence reccur, string pastorfuture)
```

Comment:
I think it would make sense to breka that method a bit up. with close to 300 lines it's quite long. For example you could break out the code in the cases into a own method. Or maybe the strategy pattern might be helpful: https://www.dofactory.com/net/strategy-design-pattern

Reply:
i believe i have made the necessary changes